src/connector/src/parser/unified/debezium.rs

```diff
@@ -158,6 +162,23 @@ pub fn parse_schema_change(
 ) -> AccessResult<SchemaChangeEnvelope> {
     let mut schema_changes = vec![];
 
+    let upstream_ddl = accessor
+        .access(&[UPSTREAM_DDL], &DataType::Varchar)?
+        .to_owned_datum()
+        .unwrap()
+        .as_utf8()
+        .to_string();
+
+    let source_id = if let Some(ScalarRefImpl::Jsonb(source_field)) =
+        accessor.access(&[SOURCE], &DataType::Jsonb)?.to_datum_ref()
+    {
+        let name: String = jsonb_access_field!(source_field, "name", string);
+        let id = name.strip_prefix(CDC_SOURCE_NAME_PREFIX).unwrap_or("0");
+        id.parse::<u32>().unwrap_or_default()
+    } else {
+        0
+    };
```

Comment:
IMO, it is more straightforward and robust to get the source id from `source_ctx.source_id` and pass it to [`parse_schema_change`](https://github.com/risingwavelabs/risingwave/blob/ba39b609cbce5b52bd59f6f6ff0a34fc4877e6e6/src/connector/src/parser/plain_parser.rs#L128) instead of extracting it from the schema change message payload.